hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/ProvidedVolumeReplicaMap.java

```diff
@@ -126,8 +129,16 @@ ReplicaInfo get(String bpid, long blockId) {
           }
           return cache.get(blockId);
         } catch (ExecutionException e) {
-          LOG.warn("Exception in retrieving ReplicaInfo for block id {}:\n{}",
-                  blockId, e.getMessage());
+          Throwable cause = e.getCause();
+          Throwable nestedCause = cause == null ? null : cause.getCause();
+          if (nestedCause != null &&
+              (nestedCause instanceof ReplicaNotFoundException ||
+                  nestedCause instanceof ReplicaRemoteFSMismatchException)) {
+            LOG.debug(e.getMessage());
```

Comment:
Shouldn't this be the warn/error?

Reply:
The method is also invoked when a new block is created (`FsDatasetImpl.createRbw`). New block creation checks in all volumes if the block is existing. A large number of warning messages are generated when new blocks are created.